scripts.js

```diff
@@ -2,4 +2,7 @@ var withButtonsClass = document.getElementsByClassName('button');
 
 for (var i = 0; i < withButtonsClass.length; i++) {
 	alert(withButtonsClass[i].innerText);
+	var numButtons = withButtonsClass.length;
```

Comment:
this variable should be outside the loop and you should use it in your if statement inside loop